doc/userManual.md

```diff
@@ -0,0 +1,183 @@
+You need to add the `sequence_id` of the command received as an input of this function.
+
+![sendCmdStatus](./images/sendCmdStatus.png)
+
+The following image shows an example of a TCP server using the library to receive commands from a TCP client.
+
+Notice that this program configures the TCP server, registers the allowed commands (`move` and `start` in this example), gets the user event reference to receive the commands, runs, and stops the TCP server.
+
+This example registers the `CmdRead` user event and waits for it.
+When the command is received it is showed in the `Command Received` indicator.
+
+![exampleCommand](./images/exampleCommand.png)
+
+After receiving the commands you can send the result of the execution, using the **sendCmdStatus.vi** with the `sequence_id` of the received command.
+
+## Using the TcpServer to receive and send events
+
+The following is a brief description of the main VIs you have to use to work with **events** in the TCP server, and an example how to implement the server.
+
+## regEvt&#46;vi
+
+The regEvt VI is used to inform the TCP server which events are registered and allowed to be processed.
+Please define all of the registered events in the `evtReg` 2-D array of strings, using the following structure:
+
+| Component Name | Event Name  |
+|:--------------:|:-----------:|
+|  component 1   |   event 1   |
+|  component 2   |   event 2   |
+
+![regEvt](./images/regEvt.png)
+
+## getInstanceEvt&#46;vi
+
+The getInstanceEvt VI is used to get the `EvtRead` user event instance, to receive an event from the TCP client.
+
+![getInstanceEvt](./images/getInstanceEvt.png)
+
+The following image shows an example of a TCP server using the library to receive events.
+
+Notice that this program configures the TCP server, registers the allowed events, gets the user event reference to receive the events, and runs the TCP server.
+
+This example registers the `EvtRead` user event and waits for it.
+When the event is received it is showed in the `Event Received` indicator.
+
+![exampleEvent](./images/exampleEvent.png)
+
+If you need to send an event to the TCP client use the **sendEvt.vi** function as the following images shows.
+
+![sendEvent](./images/sendEvent.png)
+
+## Using the TcpServer to receive and send telemetry
+
+The following is a brief description of the main VIs you have to use to work with **telemetry** in the TCP server, and an example how to implement the server:
+
+## regTel&#46;vi
+
+The regTel VI is used to inform the TCP server which telemetry is registered and allowed to be used by LabVIEW component.
+Please define all of the registered telemetry in the `telReg` 2-D array of strings,
+using the following structure:
+
+| Component Name |  Telemetry Name  |
+|:--------------:|:----------------:|
+|  component 1   |   telemetry 1    |
+|  component 2   |   telemetry 2    |
+
+![regTel](./images/regTel.png)
+
+## sendTel&#46;vi
+
+The sendTel VI is used to send the telemetry to the TCP client.
```

Comment:
Please mention the use of polymorphic vi (**convertToKeyValuePair.vi**) as above here.

Reply:
Done.